agent/src/main/java/com/thoughtworks/go/agent/service/AgentUpgradeService.java

```diff
@@ -106,6 +106,7 @@ private void validateMd5(String currentMd5, CloseableHttpResponse response, Stri
         LOGGER.debug("[Agent Upgrade Validate MD5] validating MD5 for {}, currentMD5:{}, responseMD5:{}, headerKey:{}", what, currentMd5, md5Header, agentContentMd5Header);
         if (md5Header == null) {
             LOGGER.debug("[Agent Upgrade Validate MD5] Did not find {} MD5 header in response {}.", agentContentMd5Header, response);
+            return;
```

Comment:
@JeroenOortwijn sorry for the delayed response. I think returning in case of a null md5header would remove the noise in the logs but it would also mean that the md5 is valid, which seems wrong.

Looking at the previous issues https://github.com/gocd/gocd/pull/7452, the reason for the null md5header needs to be looked into.

Reply:
Ah, it would be good to re-open issue https://github.com/gocd/gocd/issues/4469 then again?

Reply:
Re-opened.